charters/tbdex-working-group.md

```diff
@@ -0,0 +1,84 @@
+
+* Discussing, defining and implementing the tbDEX protocol specification that will best suit the needs of those who are using it in the real world
+* Discussing how updates to Web5 libraries impact tbDEX implementation, and coming up with ways to adjust the tbDEX API as needed, or influence Web5 API surfaces, if necessary
+* Discussing, defining and implementing transport specifications to send and receive tbDEX messages
+
+## Deliverables
+
+### 1. tbDEX Protocol Specification
+Description: Develop and maintain the [tbDEX Protocol Specification](https://github.com/TBD54566975/tbdex/tree/main/specs/protocol)
+
+#### Scope of Work:
+- Specification outlining types of messages and common traits in messages (structure, signature generation etc.) used by PFIs and Wallets to interact with each other to exchange value
+- Test vectors to achieve conformance across N implementations, if needed
+
+### 2. tbDEX Transport Specification
+Description: Develop and maintain the [tbDEX HTTP API Specification](https://github.com/TBD54566975/tbdex/tree/main/specs/http-api)
+
+#### Scope of Work:
+- Develop and maintain the [tbDEX HTTP API Specification](https://github.com/TBD54566975/tbdex/tree/main/specs/http-api), and additional transports in the future, outlining how to use various transports to send and receive tbDEX messages.
+  
+### 3. tbDEX Interoperability Specification
+Description: Develop and maintain the [tbDEX Interop Profile Specification](https://github.com/TBD54566975/tbdex/tree/main/specs/interop)
+
+#### Scope of Work:
+- Develop and maintain the [tbDEX Interop Profile Specification](https://github.com/TBD54566975/tbdex/tree/main/specs/interop) outlining which sets of interoperability profiles tbDEX supports at 1) tbDEX messaging layer, 2) credential / VC exchange layer
+
+### 4. tbDEX SDKs
+Description: Develop and maintain SDK implementations that conform to the tbDEX protocol and transport specifications.
+
+#### Scope of Work:
+- SDKs in [Typescript](https://github.com/TBD54566975/tbdex-js), [Kotlin](https://github.com/TBD54566975/tbdex-kt), [Swift](https://github.com/TBD54566975/tbdex-swift), [Dart](https://github.com/TBD54566975/tbdex-dart), [Go](https://github.com/TBD54566975/tbdex-go), and [Rust](https://github.com/TBD54566975/tbdex-rs)
+
+- SDKs shall demonstrate conformance to the test vectors defined by the tbDEX protocol and transport(s) specifications
+
+- SDKs shall be well-documented according to each language's conventions, as well as being surfaced through the [developer site](https://developer.tbd.website/docs/)
+
+- SDKs shall follow the guidance of the Rust Working Group in adopting a Rust core to limit developer overhead
+  
+## Success Criteria
+
+### 1. tbDEX Protocol Specification
+- A published versioned specification
+- Test vectors to be consumed across multiple tbDEX SDK implementations
+
+### 2. tbDEX Transport Specification
+- A published versioned specification
+
+### 3. tbDEX Interoperability Specification
```

Comment:
IMO these are the same spec with multiple components, and possibly more

tbDEX Spec has:
1. protocol
2. api
3. technical interop
4. pfi interop
5. trust interop (cred types, how trust is established, maintained, etc)

maybe others

Reply:
> 2. api

@decentralgabe this assumes HTTP as the only transport within the scope of the WG, yeah?

On a different matter, I'm wondering, what's in an "interop" spec which isn't in (1) and (2)?

Reply:
I vote we think of the tbDEX specification as comprehensive in and of itself. So that implies no need for dedicated interop specs, nor API specs -- the matters of both are contained within the singular spec. 

So with regards to the deliverables here in this charter I see 2: the specification and the SDK implementations.

Reply:
i think i agree with expanding the scope of what we currently call `protocol` to include trust / interop stuff. 

i still feel like the API stuff (via whatever transport, http, websocket, ftp, pigeon) belongs in separate specs, as they are "implementations" of what's outlined in the protocol spec, and can vary in the details. 

i like the idea of a developer reading the protocol that includes trust / interop stuff first, and then choosing the transport of their choice and reading the spec for that.